### PR TITLE
Context Selector

### PR DESCRIPTION
**Pull request by andresmgot**
### Description of the change

Context Selector for Clarity UI. It allows to select the cluster and the namespace used.

Peek 2020-07-08 19-33

### Applicable issues

  - ref #1354 

### Additional information

Note that the `cluster` selector is still not working. That's a WIP in a different branch.

### CODE REVIEW

File: dashboard/src/components/Header/ContextSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+  createNamespace,
+  getNamespace,
+  setNamespace,
+}: IContextSelectorProps) {
+  const dispatch = useDispatch();
+  const [open, setOpen] = useState(false);
+  const currentCluster = clusters.clusters[clusters.currentCluster];
+  const namespaceSelected = currentCluster.currentNamespace || defaultNamespace;
+  const [cluster, setCluster] = useState(clusters.currentCluster);
+  const [namespace, setNS] = useState(namespaceSelected);
```
Comment (Angelmmiguel) [inline dashboard/src/components/Header/ContextSelector.tsx:40]: I'd be consistent with the naming:

```suggestion
  const [namespace, setNamespace] = useState(namespaceSelected);
```

File: dashboard/src/components/Header/ContextSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    setOpen(false);
+  };
+
+  return (
+    <>
+      <div
+        className={`dropdown kubeapps-align-center kubeapps-dropdown ${open ? "open" : ""}`}
+        ref={ref}
+      >
+        <button className="kubeapps-nav-link" onClick={toggleOpen} aria-expanded={open}>
```
Comment (Angelmmiguel) [inline dashboard/src/components/Header/ContextSelector.tsx:69]: Glad to see `aria-expanded` here :). I'm missing here the other `aria` property to link this button with the content: `aria-haspopup`. This property must use the same value as the `role` of the element with the content that it's placed next to the button:

```suggestion
        <button className="kubeapps-nav-link" onClick={toggleOpen} aria-expanded={open} aria-haspopup="menu">
```

File: dashboard/src/components/Header/ContextSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                <span className="kubeapps-dropdown-text">{namespaceSelected}</span>
+              </div>
+            </Column>
+            <Column span={2}>
+              <div
+                className={`kubeapps-align-center angle ${open ? "angle-opened" : "angle-closed"}`}
+              >
+                <CdsIcon shape="angle" inverse={true} />
+              </div>
+            </Column>
```
Comment (Angelmmiguel) [inline dashboard/src/components/Header/ContextSelector.tsx:86]: Based on the design, I'm missing here some alignment and spacing between the items:

Design for the selector where all the items are aligned and they have more spacing

Also, any of the items is highlighted. I'd use a slightly bolder font for the selected items and a bit smaller font for the title.

File: dashboard/src/components/Header/ContextSelector.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+          </Row>
+        </button>
+        <div className="dropdown-menu" role="menu">
+          <span className="context-selector-header dropdown-menu-padding">
+            Select a cluster and a namespace to manage applications
+          </span>
+          <div className="dropdown-menu-padding" role="menuitem">
+            <CdsIcon size="sm" shape="cluster" inverse={true} />
+            <span className="kubeapps-dropdown-text">Cluster</span>
```
Comment (Angelmmiguel) [inline dashboard/src/components/Header/ContextSelector.tsx:95]: Set this `span` as a `label` and link it with the `select` via `htmlFor`

File: dashboard/src/components/Header/ContextSelector.tsx [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+                      {c}
+                    </option>
+                  );
+                })}
+              </select>
+            </div>
+          </div>
+          <div className="dropdown-menu-padding" role="menuitem">
+            <CdsIcon size="sm" shape="file-group" inverse={true} />
+            <span className="kubeapps-dropdown-text">Namespace</span>
```
Comment (Angelmmiguel) [inline dashboard/src/components/Header/ContextSelector.tsx:110]: Same about the label.

File: dashboard/src/components/Header/ContextSelector.scss [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  margin-left: -0.8rem;
+  height: 100%;
+  &-closed {
+    transform: rotate(180deg);
+    margin-left: -1rem;
+  }
+  &-opened {
+    position: relative;
+    left: 10px;
+  }
```
Comment (Angelmmiguel) [inline dashboard/src/components/Header/ContextSelector.scss:11]: Did you try to use the `rotate` property of the icon component?

File: dashboard/src/components/Header/ContextSelector.tsx [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+            <div className="clr-select-wrapper">
+              <select name="clusters" className="clr-page-size-select" onChange={selectCluster}>
+                {Object.keys(clusters.clusters).map(c => {
+                  return (
+                    <option key={`kubeapps-dropdown-cluster-${c}`} value={c}>
+                      {c}
+                    </option>
+                  );
+                })}
+              </select>
```
Comment (Angelmmiguel) [inline dashboard/src/components/Header/ContextSelector.tsx:105]: The style of the selector options is not the correct one. Clarity lefts the browser to apply their own style on those selectors:

Screenshot on how Clarity left selectors to use their native style based on the browser

File: dashboard/src/components/Header/ContextSelector.tsx (hunk 1 above)
Comment (andresmgot) [reply dashboard/src/components/Header/ContextSelector.tsx:40]: `setNamespace` is already used (by a function of the props)

File: dashboard/src/components/Header/ContextSelector.tsx (hunk 3 above)
Comment (andresmgot) [reply dashboard/src/components/Header/ContextSelector.tsx:86]: Updated:
Screenshot from 2020-07-09 17-05-17

File: dashboard/src/components/Header/ContextSelector.scss (hunk 6 above)
Comment (andresmgot) [reply dashboard/src/components/Header/ContextSelector.scss:11]: mmh, I tried but there was no `rotate` property, turns out it's called `direction`

File: dashboard/src/components/Header/ContextSelector.tsx (hunk 7 above)
Comment (andresmgot) [reply dashboard/src/components/Header/ContextSelector.tsx:105]: I don't understand what you are proposing, maybe it's a confusion? Note that I'm in linux so the style here is different:
Screenshot from 2020-07-09 17-20-37

File: dashboard/src/components/Header/ContextSelector.scss (hunk 6 above)
Comment (Angelmmiguel) [reply dashboard/src/components/Header/ContextSelector.scss]: lol 😂

File: dashboard/src/components/Header/ContextSelector.tsx (hunk 7 above)
Comment (Angelmmiguel) [reply dashboard/src/components/Header/ContextSelector.tsx:116]: Ohh I see! Completely forgot that

**Pull request merged**